pyomo/core/base/constraint.py

```diff
@@ -717,8 +717,6 @@ class Constraint(ActiveIndexedComponent):
             A dictionary from the index set to component data objects
         _index
             The set of valid indices
-        _implicit_subsets
```

Comment:
Do you need to add something to the doc class about `_anonymous_sets`?

Reply:
The catch is that `_anonymous_sets` is not *always* an attribute,.  I made the decision to try and save some time/memory and only add the attribute if it was needed.  I wasn't sure if I wanted to document an attribute that was not necessarily always there.

That may / may not be a good idea, though.  I keep wanting to try and figure out how to slotize the container classes, and this potentially further complicates that.  Maybe we should move it back to being a mandatory attribute.